include/kindyn/robot.hpp

```diff
@@ -126,6 +126,29 @@ namespace cardsflow {
                 ROS_WARN_STREAM_THROTTLE(1, "writing virtual, "
                         "you probably forgot to implement your own write function?!");
             };
+
+            /**
+             * Setter functions for gymFunctions class.
+             */
+            void setMotorState(int index1, int index2, double value);
```

Comment:
Its not clear what index1 and index2 are. My understanding is that index1 is the motor_number and index2 can be either 0 (length) or 1 (velocity). 
Since only two values are allowed for velocity we should check that for safety. Something like `assert index2 in {0, 1}`. A better solution is to have a method for each index:
`setMotorCableLength(int motor_num, double cable_lenght);` and
`setMotorCableVelocity(int motor_num, double cable_velocity);`
Also, since we are looping over all motors anyway, we may as well pass the method a vector. Something like:
`setMotorCableLengths(VectorXd cable_Lengths);` and
`setMotorCableVelocites(VectorXd cable_vels);`
Then you could use them instead of setting training_robot->Ld directly.

Reply:
done

Reply:
also adapted this for joint_state variable